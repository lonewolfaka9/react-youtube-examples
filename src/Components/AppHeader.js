import React, { Component } from 'react'

export default class AppHeader extends Component {

  constructor(props) {
    super(props);

    this.state = {
      query: "",
      suggestionArray: []// array of suggestion key word
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);


  }

  handleChange(event) {

    var query = event.target.value;
    this.setState({ query: query }, () => {
      // fetch suggestion api here
    });

  }
  handleSubmit(event) {

    var query = this.state.query;
    query = query.trim();

    if (event.key === 'Enter' && query.length > 0) {
      this.props.queryString(query);

    }
  }
  render() {
    return (
      <div style={
        {
          height: 50,
          background: "#F0F0F0",
          border: "opx solid #CCC",
          minWidth: "100vW",
          margin: "0px",
          textAlign: "cetner",
          position: "fixed",
          top:0
        }
      }>
        <input style={{
          display: "flex",
          minWidth: 300,
          height: "80%",
          margin: "0 auto",
          fontSize: 16
        }}
          type="text" id="search" placeholder="search" value={this.state.query}
          onChange={this.handleChange} onKeyPress={this.handleSubmit} />
      </div>
    )
  }
}
