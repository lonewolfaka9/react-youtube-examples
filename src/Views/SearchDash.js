import React, { Component } from 'react'

export default class SearchDash extends Component {
  render() {
    return (
      <div>
        
      </div>
    )
  }
}
