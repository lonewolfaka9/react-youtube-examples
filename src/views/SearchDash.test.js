import React from 'react';
import SearchDash from './SearchDash';
import AppHeader from '../components/AppHeader'

import { configure, shallow, mount } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import cookie from 'react-cookies'


const adapter = ReactSixteenAdapter;
configure({ adapter: new adapter.default() });
describe('Loading search View', () => {
    const historyMock = { push: jest.fn() };
    cookie.load = jest.fn(() => true)
    const searchdash = mount(<SearchDash history={historyMock} />);

    const searchdashSnap = shallow(<SearchDash history={historyMock} />);

    it('Search Page render', () => {
        expect(searchdashSnap).toMatchSnapshot();
    });



    it('initializes default `state`', () => {


        expect(searchdash.state()).toEqual({
            userAuth: true,
            query: '',
            nextPageToken: '',
            videoData: [],
            searchData: [],
            loading: "Search Something"
        });
    });

    it('Input value in AppHeader search input', () => {

        var query = "keyword";
        var appheader = searchdash.find(AppHeader)
        var searchInput = appheader.find("#search");
        searchInput.simulate('change', { target: { id: "search", name: 'search', value: query } });

        searchInput.simulate('keypress', { key: 'Enter', keyCode: 13, which: 13 });

        expect(appheader.state().query).toEqual(query); // query updated in app header

        expect(searchdash.state().query).toEqual(query);

    });


});
