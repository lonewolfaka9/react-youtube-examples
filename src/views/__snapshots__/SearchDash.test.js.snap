// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading search View Search Page render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchDash
    history={
      Object {
        "push": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <AppHeader
          queryString={[Function]}
        />,
        <div
          className="col two"
        >
          Search Something
        </div>,
        <div
          className="row"
        >
          <div
            className="col  one"
          >
                
                
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "queryString": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search Something",
          "className": "col two",
        },
        "ref": null,
        "rendered": "Search Something",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col  one"
          >
                
                
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "    ",
              Array [],
              "    ",
            ],
            "className": "col  one",
          },
          "ref": null,
          "rendered": Array [
            "    ",
            "    ",
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <AppHeader
            queryString={[Function]}
          />,
          <div
            className="col two"
          >
            Search Something
          </div>,
          <div
            className="row"
          >
            <div
              className="col  one"
            >
                  
                  
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "queryString": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Something",
            "className": "col two",
          },
          "ref": null,
          "rendered": "Search Something",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col  one"
            >
                  
                  
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "    ",
                Array [],
                "    ",
              ],
              "className": "col  one",
            },
            "ref": null,
            "rendered": Array [
              "    ",
              "    ",
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
